katas/012_vector_search.md

```diff
@@ -0,0 +1,220 @@
+                 "normalize": false,
+                 "cache" : "embeddings"
+               }
+           }
+  }'
+fi
+```
+
+The last piece of script to take a look at is the following one, which defines vector enabled relevancy algorithms using ParamSets. There are 4 of them, one for boost by image (img), one for match by image, one for boost by text (txt) and one for match by text:
+
+```bash
+if $vector_search; then
+  log_minor "Defining Vector enabled relevancy algorithms using ParamSets."
+
+  curl --user solr:SolrRocks -X POST http://localhost:8983/solr/ecommerce/config/params -H 'Content-type:application/json'  -d '{
+    "set": {
+      "querqy_boost_by_img_emb":{
+        "defType":"querqy",
+        "querqy.rewriters":"embimg",
+        "querqy.embimg.topK": 100,
+        "querqy.embimg.mode": "BOOST",
+        "querqy.embimg.boost": 10000,
+        "querqy.embimg.f": "product_image_vector",
+        "qf": "id name title product_type short_description ean search_attributes",
+        "querqy.infoLogging":"on",
+        "mm" : "100%"
+      }
+    },
+    "set": {
+      "querqy_match_by_img_emb":{
+        "defType":"querqy",
+        "querqy.rewriters":"embimg",
+        "querqy.embimg.topK":100,
+        "querqy.embimg.mode": "MAIN_QUERY",
+        "querqy.embimg.f": "product_image_vector",
+        "qf": "id name title product_type short_description ean search_attributes",
+        "querqy.infoLogging":"on",
+        "mm" : "100%"
+
+      },
+      "querqy_boost_by_txt_emb":{
+        "defType":"querqy",
+        "querqy.rewriters":"embtxt",
+        "querqy.embtxt.topK": 100,
+        "querqy.embtxt.mode": "BOOST",
+        "querqy.embtxt.boost": 10000,
+        "querqy.embtxt.f": "product_vector",
+        "qf": "id name title product_type short_description ean search_attributes",
+        "querqy.infoLogging":"on",
+        "mm" : "100%"
+      },
+      "querqy_match_by_txt_emb":{
+        "defType":"querqy",
+        "querqy.rewriters":"embtxt",
+        "querqy.embtxt.topK":100,
+        "querqy.embtxt.mode": "MAIN_QUERY",
+        "querqy.embtxt.f": "product_vector",
+        "qf": "id name title product_type short_description ean search_attributes",
+        "querqy.infoLogging":"on",
+        "mm" : "100%"
+      }
+    },
+  }'
+fi
+```
+
+## Understanding Vector Search
+
+Let's dive into Vector Search!
+
+What we refer to as "vector search" is also known as "neural search", or "dense vector search", or sometimes "cognitive search", among other names.
+
+In traditional search, sparse vectors are used in bit sets.  A sparse vector contains a list of numbers and a lot of those values are going to equal to zero, and a few non-zero values.  This is typical in traditional search, where there are a lot of bits (thousands) dedicated to whether or not a document does or does not meet some criteria.  A silly and overly simplified example of a sparse vector might look something like the following scenario:
```

Comment:
"are going to equal to zero"
can be modified to "are going to be equal to zero"